### PR TITLE
support ssh-agent

### PR DESCRIPTION
**Pull request by takaishi**
When we add private key to ssh-agent, we want to use it in sshc.

### CODE REVIEW

**Comment by takaishi** [comment]
@k1LoW Hi, I request to review.

File: sshc.go [hunk 1]
```diff
@@ -197,7 +198,24 @@ func (c *Config) DialWithConfig() (*ssh.Client, error) {
 		}
 		fmt.Println("")
 	}
-	auth = append(auth, ssh.PublicKeys(signer))
+
+	if sshAuthSockExists() {
```
Comment (k1LoW) [inline sshc.go]: I think we need to check `ForwardAgent` too.

```golang
if c.Get(host, "ForwardAgent") == "yes" && sshAuthSockExists() {
// ...
}
```

File: sshc.go [hunk 2]
```diff
@@ -13,7 +14,7 @@ import (
 
 	"github.com/ScaleFT/sshkeys"
 	"github.com/kevinburke/ssh_config"
-	homedir "github.com/mitchellh/go-homedir"
+	"github.com/mitchellh/go-homedir"
```
Comment (k1LoW) [inline sshc.go]: Thank you !!!

File: integration_test.go [hunk 3]
```diff
@@ -22,6 +24,39 @@ func TestSSH(t *testing.T) {
...
+		got, err := getHostname(tt.hostname)
+		if err != nil {
+			t.Fatal(err)
+		}
+		want := fmt.Sprintf("%s\n", tt.hostname)
+		if got != want {
+			t.Fatalf("want = %#v, got = %#v", want, got)
+		}
+	}
+
```
Comment (k1LoW) [inline integration_test.go]: If you fix https://github.com/k1LoW/sshc/pull/6#discussion_r269094690 , could you fix tests too ?

File: sshc.go (hunk 1 above)
Comment (takaishi) [reply sshc.go]: I agree, need to check. I fixed in https://github.com/k1LoW/sshc/pull/6/commits/e9ad09a12781822de6c29f15716268a8362d1736 and https://github.com/k1LoW/sshc/pull/6/commits/c2ed0d3621dda5b21af9cf3c18f736a81c62c6b5 !

File: integration_test.go (hunk 3 above)
Comment (takaishi) [reply integration_test.go]: I fixed test in https://github.com/k1LoW/sshc/pull/6/commits/c2ed0d3621dda5b21af9cf3c18f736a81c62c6b5 .

File: .travis.yml [hunk 4]
```diff
@@ -14,6 +14,7 @@ install:
   - sudo pip install codecov
   - docker-compose up -d
 script:
+  - chmod 600 ./testdata/id_rsa
```
[the remark this answers is not in the document]
Comment (takaishi) [reply .travis.yml]: Yes, we available remove this, I want .

**Comment by k1LoW** [comment]
@takaishi Thank you !!!

**Pull request merged**